.github/workflows/build.yml

```diff
@@ -1,34 +1,208 @@
-name: Build
-on:
-  pull_request:
-    branches: [master]
-  push:
-    branches:
-      - master
+name: CD Pipeline
+
+on: [push]
+
 jobs:
-  test:
-    name: Test Suite
+  build-nix:
+    env:
+      IN_PIPELINE: true
+    runs-on: ${{ matrix.os }}
+    if: github.ref == 'refs/heads/main'
```

Comment:
Would this need to be `refs/heads/master` for us?

Equally, is this effectively the same as the old config. (which is a bit more obvious, IMHO)?

```yaml
  push:
    branches:
      - master
```